hredis/redigo/redigo.go

```diff
@@ -42,15 +44,73 @@ func WaitForAvailability(url string, d time.Duration, f WaitFunc) (bool, error)
 	}
 }
 
+// make this a var, so we can change this for testing.
+var redisDialURL = redis.DialURL
+
 // NewRedisPoolFromURL returns a new *redigo/redis.Pool configured for the supplied url
 // The url can include a password in the standard form and if so is used to AUTH against
 // the redis server
-func NewRedisPoolFromURL(url string) (*redis.Pool, error) {
+func NewRedisPoolFromURL(rawURL string, altPasses ...string) (*redis.Pool, error) {
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		mu       sync.Mutex
+		password string
+	)
+
+	// strip and save password
+	if pass, ok := u.User.Password(); ok {
```

Comment:
What do you think about using a single slice of passwords instead of a password and an optional slice of extra passwords. I believe using just a single slice would make some of the logic a little cleaner. I could be wrong though.

Reply:
I'm all for the simplicity of a single slice of passwords, but the common usage here is that the URL contains the password (e.g. on Heroku), which means that _someone_ will have to do this work...

Reply:
I was trying to suggest internally using a slice of passwords. In the common case there would only be one element in the slice. If there are 2 alternate passwords then there would be 3 elements in the slice. This way the dial() function could range on the slice and not have to track a seperate password.